src/components/UserRolesInfoModal/UserRolesInfoModal.js

```diff
@@ -0,0 +1,270 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled, { css } from 'styled-components/macro'
+
+import { ButtonSecondary } from '../generic/buttons'
+import { IconCheck, IconClose } from '../icons'
+import { Table, Td, Tr, TableOverflowWrapper } from '../generic/Table/table'
+import language from '../../language'
+import theme from '../../theme'
+import Modal, { RightFooter } from '../generic/Modal/Modal'
+
+const Thead = styled.th`
+  text-align: center;
+  background-color: ${theme.color.primaryColor};
+  color: white;
+  padding: ${theme.spacing.small} ${theme.spacing.medium};
+  vertical-align: top;
+  span {
+    white-space: nowrap;
+  }
+  small {
+    display: block;
+  }
+`
+
+const Tcell = styled(Td)`
+  ${(props) =>
+    props.cellWithText
+      ? css`
+          text-align: left;
+        `
+      : css`
+          text-align: center;
+        `};
+`
+
+const TcellWithIconClose = styled(Tcell)`
+  color: ${theme.color.cautionColor};
+`
+const UserRolesInfoModal = ({ isOpen, onDismiss }) => {
+  const modalContent = (
+    <TableOverflowWrapper>
+      <Table>
+        <thead>
+          <Tr>
+            <Thead>Accessible Information</Thead>
```

Comment:
How strict are we about using the language file these days? Should most of this be coming from the language file?

Reply:
for this feature, I followed the same pattern as the data sharing info modal (most of the text in the js file). I think if we want to be more strict on it, we should do a batch update of files as a separate task